dcos-log/api/server.go

```diff
@@ -2,60 +2,110 @@ package api
 
 import (
 	"crypto/tls"
+	"crypto/x509"
+	"errors"
 	"fmt"
+	"io/ioutil"
 	"net/http"
+	"os"
 	"time"
 
 	"github.com/Sirupsen/logrus"
 	"github.com/coreos/go-systemd/activation"
-	"github.com/dcos/dcos-go/dcos"
+	"github.com/dcos/dcos-go/dcos/nodeutil"
 	"github.com/dcos/dcos-go/jwt/transport"
 	"github.com/dcos/dcos-log/dcos-log/config"
 )
 
-func newDCOSUtil(cfg *config.Config, rt http.RoundTripper) (dcos.Util, error) {
-	if cfg.FlagNoAuth {
+func newNodeInfo(cfg *config.Config, rt http.RoundTripper) (nodeutil.NodeInfo, error) {
```

Comment:
seems like it be slightly more efficient for this func to accept a **copy** of an `http.Client` that it then would customize the transport for. you wouldn't then need to, for example, parse the timeout twice